### ISSUE TITLE
ocean SSS restoring not working correctly

### ISSUE THREAD

**Issue by mark-petersen** [MPAS-Dev/MPAS-Model#120]
From @maltrud: The value of the climatological SSS field that is used for SSS restoring is not being updated correctly. i ran a E3SM G-case for 15 months and while the first month agreed with the read-in climatology file, subsequent months didn't change much--certainly no seasonal cycle.

i am currently testing a version of the code from before the following commit to see if that may be the cause.

commit d2a7f499a5c56b0f413161489fb810811b5e9096
Author: Mark Petersen <mpetersen@lanl.gov>
Date:   Thu Nov 16 10:09:34 2017 -0700

    Only a single forcing group linked list should exist per core.
    
    These were put in by Adrian in PR 1333, but then removed
    inadvertenly in PR 1427

**Comment by maltrud** [comment]
i just tested a case from a version of ACME i cloned Sep 28, 2017, and the SSS restoring is working correctly.

**Comment by maltrud** [comment]
an interesting twist:  i just checked the fields read in when BGC is enabled, and they seem to be fine.  for example the iron flux has the expected magnitude and seasonal cycle.  makes me wonder if the problem is related to other changes made to the SSS restoring routines, and not the forcing framework changes.

**Comment by vanroekel** [comment]
@maltrud and @mark-petersen I think I know what the problem is here.  When the following line is called
```
call MPAS_forcing_get_forcing(surfaceSalinityForcingGroup, &
                'surfaceSalinityMonthlyClimatology', streamManager, dt)
```
data is read, and the forcing clock is advanced by ```dt```.  This is correct if forcing data is read every time step, but salinity is read only every ```config_surface_salinity_monthly_restoring_compute_interval```, for most runs this is 1 day.  So for a 15 month run at EC60to30, the forcing clock only advances 450 time steps (or about 3 days for a 10 minute time step).

To solve this, what if we set these two lines (L303 and 304 of mpas_ocn_tracer_surface_restoring.F)

**Comment by maltrud** [comment]
thanks for looking into this, @vanroekel .  the forcing stuff is still a mystery to me, but your proposal sounds good.  lets give it a try.

**Comment by akturner** [comment]
Hi,


So Data doesn't need to be read everytime step, but the system assumes you want a new interpolated value everytime step. Thats why the clock advances. Is this not how its being used in the ocean model?


Cheers

Adrian

________________________________

**Comment by vanroekel** [comment]
Ade,

  Yes, we are only accessing a new data value once per day instead of once per timestep to make the routine more efficient.  I think to fix this issue we can advance the forcing clock by one day instead of dt.  Do you think that would work?


Regards,

Luke

________________________________

**Comment by akturner** [comment]
Hi,


Off the top of my head, yes that should work. Let me dig around the code to confirm.


Cheers

Ade

________________________________

**Comment by akturner** [comment]
Yeah, that should be fine.


Ade

________________________________

**Comment by mark-petersen** [comment]
Fixed by #138
